pkg/provider/rule.go

```diff
@@ -72,6 +75,20 @@ func appendMethodFilter(matchParts []string, match *specs.HTTPMatch) []string {
 	return matchParts
 }
 
+func appendHeaderFilter(matchParts []string, match *specs.HTTPMatch) []string {
+	var rules []string
+
+	for name, value := range match.Headers {
```

Comment:
Questions:

- Is the `httpHeaders` type necessary?
- Should we rename `ServiceTrafficTarget#Specs` to `ServiceTrafficTarget#Rules`?
- Do you think that it could make sense to create the `rules` array with the right size? 

```
rules := make([]string, len(match.Headers))
```